<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>957</width>
    <height>786</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Road sign detection</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="OpenCVWidget" name="openCVView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>957</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="minimumSize">
    <size>
     <width>338</width>
     <height>160</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>160</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QLabel" name="consoleLbl">
         <property name="text">
          <string>Output console:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="outputConsole">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="loadImgBtn">
         <property name="text">
          <string>Load image...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="loadVideoBtn">
         <property name="text">
          <string>Load video...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="playPauseBtn">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Pause</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="minimumSize">
    <size>
     <width>277</width>
     <height>562</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Shape type:</string>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <item row="0" column="0" colspan="2">
         <widget class="QRadioButton" name="regPolyRadio">
          <property name="text">
           <string>Regular polygon</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Number of sides:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="numSides">
          <property name="value">
           <number>8</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QRadioButton" name="rectRadio">
          <property name="text">
           <string>Rectangle</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Aspect ratio:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="horizAspect">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="vertAspect">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>3</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Shape sizes:</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Minimum:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="minSize">
          <property name="text">
           <string>6</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Maximum:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="maxSize">
          <property name="text">
           <string>15</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Step:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="step">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Rotation:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="rotation">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Output filtering:</string>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Shape limit:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="shapeLimit">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Min. votes:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="voteReq"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="goBtn">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Go!</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="saveAsVideo">
       <property name="text">
        <string>Save output as video</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="useCCOR">
       <property name="text">
        <string>Use CCOR filter (experimental)</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenCVWidget</class>
   <extends>QGraphicsView</extends>
   <header>opencvwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>loadVideoBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadVideo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadImgBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playPauseBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>togglePause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>loadVideo()</slot>
  <slot>loadImage()</slot>
  <slot>togglePause()</slot>
 </slots>
</ui>
